android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +82,160 @@ public void init(String siteKey, String touchpointUri, String apiKey,
+          Log.e(NAME, "[Thunderhead] Send Properties SDK Error: " + error.getErrorMessage());
+        } catch (OneAPIError error) {
+          notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Api Error: " + error.getErrorMessage());
+        } catch (CompletionException error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Completion Error: " + error.getLocalizedMessage());
+        } catch (Exception error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction Error: " + error.getLocalizedMessage());
+        }
+      } else {
+        try {
+          OneResponse response;
+          response = One.sendPropertiesLegacySupport(THROW_ERRORS, sendPropertiesRequest).join();
+          WritableNativeMap responseMap = responseObjectToReadableMap(response);
+          notifyResult(promise, responseMap);
+        } catch (ExecutionException error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Completion Error: " + error.getCause());
+        } catch (OneSDKError error) {
+          notifyProblem(promise, Integer.toString(error.getSystemCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties SDK Error: " + error.getErrorMessage());
+        } catch (OneAPIError error) {
+          notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Properties Api Error: " + error.getErrorMessage());
+        } catch (Exception error) {
+          notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+          Log.e(NAME, "[Thunderhead] Send Interaction Error: " + error.getLocalizedMessage());
+        }
+      }
+    });
   }
 
   @ReactMethod
-  public void sendResponseCode(String responseCode, String interaction) {
-    final OneResponseCodeRequest responseCodeRequest = new OneResponseCodeRequest.Builder()
-      .responseCode(new OneResponseCode(responseCode))
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
-      .build();
-    One.sendResponseCode(responseCodeRequest).enqueue(null);
+  public void sendResponseCode(String interactionPath, String responseCode, final Promise promise) {
+    executor.submit(() -> {
+      final OneResponseCodeRequest responseCodeRequest = new OneResponseCodeRequest.Builder()
+        .responseCode(new OneResponseCode(responseCode))
+        .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
+        .build();
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
Why the dupe of try catch?  You had a much more concise setup before

Reply:
It's because `CompletionException` is only available on Android N+

Reply:
If there's a cleaner way let me know